### PR TITLE
Cleanup s3 local test and add polling coverage for model version updates from S3

### PR DESCRIPTION
**Pull request by CoderHam**
The polling test passes i.e. adding a new version directory to a model triggers the polling and the new version is loaded as expected.

### CODE REVIEW

File: qa/L0_s3_local/test.sh [hunk 1]
```diff
@@ -93,56 +97,60 @@ awslocal $ENDPOINT_FLAG s3 rm s3://demo-bucket1.0 --recursive --include "*" && \
...
-fi
-set -e
-
-kill $SERVER_PID
-wait $SERVER_PID
-
-# Test with IP
-SERVER_ARGS="--model-repository=s3://127.0.0.1:4572/demo-bucket1.0 --model-control-mode=explicit"
-SERVER_LOG="./inference_server_ip.log"
+# Test with Polling
```
Comment (GuanLuo) [inline qa/L0_s3_local/test.sh:151]: Should test the same in explicit mode

**Review by GuanLuo — CHANGES_REQUESTED** [review]
I don't see changes on the source code, so it is polling from the correct cloud repository?

File: qa/L0_s3_local/test.sh (hunk 1 above)
Comment (CoderHam) [reply qa/L0_s3_local/test.sh:151]: Already tested in explicit mode in L0_storage_S3

**Comment by CoderHam** [comment]
> I don't see changes on the source code, so it is polling from the correct cloud repository?

Yes that is correct. It appears my earlier observation was incorrect due to sleep that was too low.

**Review by GuanLuo — APPROVED** [review]
> Already tested in explicit mode in L0_storage_S3

Can you create a follow up ticket to group the S3 tests into one? I realize we have different test folders that just test different flavors of S3, which can be hard to track. For instance, with this change, the generic model loading functionality tests is across L0_s3_local and L0_storage_S3, which can be confusing. What we can do is having the single test where the first half tests the generic cases against the common S3 usage, and the second half would be sanity check on other flavors are handled properly

File: qa/L0_s3_local/test.sh [hunk 2]
```diff
@@ -44,15 +44,19 @@ CLIENT_LOG="./client.log"
...
 DATADIR="/data/inferenceserver/${REPO_VERSION}/qa_model_repository"
+BACKENDS="graphdef libtorch onnx plan savedmodel"
 
-MODELDIR="models"
-rm -rf ./$MODELDIR && mkdir $MODELDIR
-cp -r $DATADIR/*_float32_float32_float32 $MODELDIR/.
+rm -rf models && mkdir models
+for BACKEND in $BACKENDS; do
+    cp -r $DATADIR/${BACKEND}_float32_float32_float32 models/.
+    # Remove version policy from config.pbtxt
```
Comment (tanmayv25) [inline qa/L0_s3_local/test.sh:52]: Why is this needed?

**Comment by CoderHam** [comment]
> Can you create a follow up ticket to group the S3 tests into one?

Sure

File: qa/L0_s3_local/test.sh (hunk 2 above)
Comment (CoderHam) [reply qa/L0_s3_local/test.sh:52]: The version policy allows 1 and 3 but we want to autoload 4 when and unload 3 when 4 is added to model repo and polling is enabled

**Review by tanmayv25 — APPROVED** [review]
